Sources/Scheduler.swift

```diff
@@ -162,11 +159,17 @@ public final class UIScheduler: Scheduler {
 		// already enqueued, we can skip scheduling and just execute directly.
 		if queued == 1 && DispatchQueue.getSpecific(key: UIScheduler.dispatchSpecificKey) == UIScheduler.dispatchSpecificValue {
 			actionAndDecrement()
+			return nil
 		} else {
-			DispatchQueue.main.async(execute: actionAndDecrement)
-		}
+			let disposable = AnyDisposable()
 
-		return disposable
+			DispatchQueue.main.async {
+				guard !disposable.isDisposed else { return }
```

Comment:
The only change is that `schedule` that is done immediately no longer returns a disposable, aligning to the semantic of `ImmediateScheduler`. Per the `Scheduler` contract, it is wrong to assume `schedule` would always return a cancellation token anyway.